### ISSUE TITLE
Several Shortcuts stop Working in 1.8.4

### ISSUE THREAD

**Issue by FlashBlaze** [Zettlr/Zettlr#1560]
## Description
Pressing `CTRL+A` to select all shifts the cursor at the beginning of the line instead of selecting everything

## Reproducing
Create a file, write some text and press `CTRL+A`

## Expected behaviour
Pressing `CTRL+A` should select everything.

## Platform
 - OS and version: Windows 10 20H2
 - Zettlr Version: 1.8.4

## Additional information
Used to work fine on v1.8.3 and previous versions

**Comment by nathanlesage** [comment]
Would you mind creating a GIF demonstrating what is happening there?

**Comment by FlashBlaze** [comment]
https://user-images.githubusercontent.com/13968549/103079136-5333a980-45f9-11eb-97a9-52c4735a9c04.mp4


`CTRL+V` is also not working after copying using `CTRL+C`. I can paste using the option displayed after right-clicking which confirms that the text is being copied using `CTRL+C`.

**Comment by noah-arroyo** [comment]
I, too, am experiencing these bugs: 

- `CTRL+A` = cursor jumps to beginning of that line.
- `CTRL+C` = copies selected content (as it should, but...)
- `CTRL+V` = doesn't paste.

My operating system: Windows 10
My version Zettlr: 1.8.4

***

In addition, I'm experiencing the following:

- `CTRL+V` = cursor jumps down 33 lines (not paragraphs) on the document.
- `CTRL+N` = cursor jumps down one line (not paragraph) on the document.
- `CTRL+Z` = doesn't undo, within document. Instead, undoes the most recently typed characters in the file directory filter.
- `CTRL+Y` = doesn't redo, within document. Instead, redoes the most recently typed characters in the file directory filter

Including these additional bugs here because I suspect they might be related. On this comp, this is the first time I've encountered these bugs, and I have consistently updated to the most recent version of Zettlr for ~9 months.

**Comment by BYWL** [comment]
me too
windows 10
1.8.4

**Comment by nathanlesage** [comment]
> Including these additional bugs here because I suspect they might be related.

Very likely, yes.

You know what? I almost think that there might be a wrong input mode enabled in your installs, because it's really hard to imagine ALL of these shortcuts failing at the very same time. Could that be? I recently removed the VIM mode, and it might be that my own migration code didn't work perfectly so that your installs are, behind the scenes, stuck in VIM mode even though they may show you something different …

Would you mind confirming that by backing up your `config.json` and just trying to edit again?

**Comment by noah-arroyo** [comment]
> Would you mind confirming that by backing up your config.json and just trying to edit again?

@nathanlesage Apologies, this is just beyond my current know-how. Is there a good explainer that you could link to?

**Comment by nathanlesage** [comment]
@noah-arroyo 

It might help to clean out the application data, as some errors relate to corrupted files by the underlying framework (Electron) themselves. By cleaning out the app data, we can exclude this possibility. Here's how you safely clean out the app data:

1. First ensure that the application is closed (in some error cases, there is a "zombie" process, which you can close from within your Task Manager or Activity Monitor)
2. Make sure to optionally backup these files:
    - `stats.json` (your writing statistics)
    - `config.json` (the app settings)
    - `custom.css` (your custom CSS, if any)
    - `tags.json` (your colour-coded tags)
    - `targets.json` (your writing targets)
    - `user.dic` (your own custom dictionary)
3. For informative purposes, it might also make sense to backup the `logs`-folder, which contains logs from the application.
4. Select everything in this directory and delete it.
5. Restart the application. Some of these files will be re-created.

The application data path is …

* … on Windows: `C:\Users\<your username>\AppData\Roaming\zettlr`
* … on macOS: `/Users/<your username>/Library/Application Support/zettlr`
* … on Linux: `/home/<your username>/.config/zettlr`

***

_I like auto-replies …_

**Comment by nathanlesage** [comment]
Update: In #1582 it appears this started in 1.8.3

So if anyone is up to reading a lot of commits, please go forward seeing what might have caused it!

**Comment by noah-arroyo** [comment]
@nathanlesage Followed your instructions and deleted all in that directory. Problem is unchanged. Bummer.

**Comment by nathanlesage** [comment]
@noah-arroyo Thank you for confirming that it's not my migratory code!

**Comment by femok** [comment]
I experienced this issue only on one of my machines after installing zettlr for the first time on two machines (both Windows 10
and zettlr 1.8.4). So I guess @nathanlesage is right with the 'corrupted files'-thesis. However, until yet I couldn't resolve the error.

**Comment by noah-arroyo** [comment]
Right. Like @femok, I have Zettlr 1.8.4 installed on two comps (both Windows 10) but this bug is happening on only one of them.

@nathanlesage Therefore, it seems like both @femok and I are in ideal positions to locate what's different between our comps or installs, to help you figure out what's causing the bug. I'm no tech wiz, so I'd probably need some guidance.

**Comment by nathanlesage** [comment]
@noah-arroyo Oh? That sounds interesting …

Hm … if the app is installed on two computers, and on one it's working, while on the other it isn't, the problem must be located in the application data. The application itself is obviously the same on both machines (as you downloaded the exact same installer for both computers).

Soooo, I think the course of action could first look like this:

1. Go to your application data folder (the Roaming one I referenced above)
2. One-by-one go through the following files and look for any difference:
    - config.json (Settings file)
    - stats.json (your writing statistics)
    - custom.css (your custom CSS, if any)
    - tags.json (your colour-coded tags)
    - targets.json (your writing targets)
    - user.dic (your own custom dictionary)
3. If the only difference are the paths in the config.json that are being loaded, this might indicate that in there something odd is happening. If this is so, try to remove one path after another (manually from the config.json file, while Zettlr is NOT running) UNTIL the shortcuts start working again. Then we have the culprit directory (assuming that is in fact the cause).

If nothing works, report back with anything that you have encountered, even if it doesn't seem special!

***

**To have an easier job looking at the JSON files, you can paste them into this online converter which will apply syntax highlighting and make the file look good: https://jsonformatter.curiousconcept.com/**

The JSON in the configuration files of Zettlr are stored without spaces and line breaks because this saves a little space, so you're not going to break anything :)

**Comment by noah-arroyo** [comment]
Excellent. Thanks @nathanlesage 

I'm pretty sure I can handle this by tonight. (Right now is ~noon, where I live.)

**Comment by john-r-wolfe** [comment]
Hi All,

I'm experiencing the exact same issues. On my personal PC I'm having no trouble. On my ωork laptop I am having the same issues as Noah where CTRL+A, CTRL+V, and CTLR+Z do not work as intended.

I inspected the differences between the two config.json files, and the only differences were the open paths and the "uuid". I cleared the paths from both config.json, set the root directory to Desktop, and created a new file to test the shortcuts - the issues persisted.

I noticed that on my personal PC I installed 'for all users' so the files are in C:\Program Files, whereas for the other installation I had selected 'for my user' so my install files are in C:\Users\myusername\AppData\Local\Programs - could this perhaps be meaningful? @noah-arroyo are you by chance in the same situation?

Thank you.

**Comment by noah-arroyo** [comment]
@jwolf112 Regarding install location:

For me, Zettlr is installed on both comps 'for my user', so that's not it.

(Just for kicks, I tried doing a fresh install 'for all users' on the misbehaving comp, but the bug persisted.)

My more robust message on troubleshooting is below this one...

**Comment by noah-arroyo** [comment]
Okie dokie. Keep in mind that I'm only tech-conversant, not an expert by any measure.

Below is what I noticed.

**tl;dr** - I couldn't figure out what's causing the bug. I'm making my files available to you via the cloud -- see end of this dialogue for instructions to access.

***

NOTE: Within these explanations, I'll use the following notation:

{B} = bad comp
{G} = good comp

***

{G}'s' directory (C:\Users\narro\AppData\Roaming\zettlr) contains one folder that {B}'s' doesn't: `databases`. In it, are these files:
* `Databases.db`
* `Databases.db-journal`

I tried copying {G}'s `databases` folder into {B}'s directory, but...

FAIL: The bug persisted.

***

Tell me if I followed your instructions correctly:

While Zettlr was closed... In {B}'s file, `config.json`, I deleted the paths listed within brackets, for `"openPaths":[]`. You'd suggested I delete each path one-by-one, but I opted for the sledgehammer approach and deleted them all just to see if it'd kill the bug. (If so, then I'd put them back and try them one-by-one.)

FAIL: The bug persisted.

In that file, there are two other sections of code that contain the word "Path": `"dialogPaths":{}` and `"defaultSaveImagePath"`

For _the former_, I again tried deleting all the listed paths to see if it killed the bug.

FAIL: The bug persisted.

For _the latter_, I did nothing becuase it appeared to be identical to the code in {G}'s config.json file.

***

In {B}, I work with files via Zettlr in a directory within my offline-synced cloud storage -- I use Box, a competitor to Dropbox, so that my files are constantly synced to the cloud and version-controlled.

I wondered whether _something, ANYthing_ about this approach was confounding Zettlr. So I pointed {B}'s Zettlr at Windows' `documents` folder (the default save location on a fresh Zettlr install, if I recall). I created a new file there and worked with it, but...

FAIL: The bug was there too.

***

I've actually never used {G} for anything -- when Zettlr 1.8.4 came out I just installed it and left it alone. Therefore, unlike {B}, many of the files that you flagged for inspection were blank or nearly so. I'm not sure what to do with that, in order to locate the bug.

***

That's why I'm making copies of the Roaming directories (for both {G} and {B}) available to you for inspection. I'm direct-messaging you in Twitter with the links to where they'll live in Box, along with a password to access.

I'll be on standby to try whatever you suggest, on my comps. It may take me a few hours to respond but I'll do my best.

**Comment by nathanlesage** [comment]
Thanks to @noah-arroyo we now have ruled out any malformed files, so the bug MUST be in the code base, that is:

This list here contains the bug SOMEWHERE: https://github.com/Zettlr/Zettlr/compare/v1.8.3...v1.8.4

So here's my offer: I'll release 1.8.5 on Friday, maybe it was just a "happy little accident", and 1.8.5 automatically fixes it. Otherwise, we have to sift through that list …

**Comment by nathanlesage** [comment]
Short update: couldn't find anything in those commits, so maybe it's an error in the build itself, I'm now betting all my hopes on the 1.8.5 release …

**Comment by femok** [comment]
@nathanlesage and @noah-arroyo 

So, here comes a short comparison of my installations: 

![image](https://user-images.githubusercontent.com/25936335/103795324-209c9e80-5046-11eb-8522-80bc7c58b169.png)
The working installation is on my t460s, the installation on my surface is the buggy one. 


**config.json:** 
![image](https://user-images.githubusercontent.com/25936335/103795515-60fc1c80-5046-11eb-982f-b58a728bff6e.png)

These files are different. I reformatted the files as suggested, here comes a detailed comparison: 
![image](https://user-images.githubusercontent.com/25936335/103795749-a7517b80-5046-11eb-9fee-de9c1b960e3b.png)
![image](https://user-images.githubusercontent.com/25936335/103795780-b2a4a700-5046-11eb-9b56-1bf91e246328.png)
![image](https://user-images.githubusercontent.com/25936335/103795813-bb957880-5046-11eb-89b7-f8f94aec33e4.png)


**custom.css:**
I played around with customization of my zettlr, but I guess that's not causing the problem. 
![image](https://user-images.githubusercontent.com/25936335/103795877-d2d46600-5046-11eb-9687-b9d18a7a7b45.png)


**stats.json:**
![image](https://user-images.githubusercontent.com/25936335/103796014-fe575080-5046-11eb-87bc-bd719551a422.png)


As this might be helpful: 
I'm using a cloud-service to sync my *.md files, both directories are on the same path (including username). 

I apologize for the delay and sorry for the bad image quality. However, hope this helps.

**Comment by nathanlesage** [comment]
@femok Thank you very much!

Two things I notice:

1. Your not working installation is on ARM, so in the worst case there's a bug in the install-maker so that the installers are faulty. That might have been just a temporary bug
2. There are two absolute file paths in the openFiles array which does not seem right

**Comment by femok** [comment]
@nathanlesage Great, thank you!

according 1.:
What indicates the not working installation is on ARM? As my surface is a few years old, it - technically - couldn't be on ARM. 
Here the hardware specs: CPU: Intel Core i5-8350U // RAM: 8GB // SSD: 256GB // OS: Win10

according 2.:
I guess you're referring to the lines 8-10 in config.json (orange background) on the surface installation? As the lines 8 and 9 are different, can I just remove them without any risk? On both machines, there should be only two folders loaded, which are the zettlr-tutorial folder and the cloud-folder with my *.md files.

**Comment by KarlClinckspoor** [comment]
I have the same problem, Win 8.1 Zettlr 1.8.4, which I downloaded with the (new?) in-app upgrade button, cool. Reset the settings and nothing changed. Here's to hoping 1.8.5 fixes the problem. As a temporary workaround, [I've found](https://en.wikipedia.org/wiki/Table_of_keyboard_shortcuts) that the following classic shortcuts still work:

Copy: Ctrl+Ins
Paste: Shift+Ins
Undo: Alt+Backspace
Redo: Alt+Shift+Backspace

**Comment by nathanlesage** [comment]
@femok 

> What indicates the not working installation is on ARM? As my surface is a few years old, it - technically - couldn't be on ARM.

My mistake!

> I guess you're referring to the lines 8-10 in config.json (orange background) on the surface installation? As the lines 8 and 9 are different, can I just remove them without any risk?

I would remove the paths. There are only numbers supposed to be in there.

**Comment by KarlClinckspoor** [comment]
Just updated to 1.8.5 and can confirm the shortcuts are working again! Tested Copy, Paste, Select All, Undo, Redo.

**Comment by FlashBlaze** [comment]
@nathanlesage can confirm. 1.8.5 solved all the issues. Thanks a lot!

**Comment by nathanlesage** [comment]
Oh my God, yes! I am really, really happy!!! :) Thank you all!

**Comment by noah-arroyo** [comment]
Me too!!! 1.8.5 works like a dream.

**Comment by femok** [comment]
Yes, everything is working agin :) Thx all, especially @nathanlesage !
